common/src/main/java/com/microsoft/identity/common/exception/IntuneAppProtectionPolicyRequiredException.java

```diff
@@ -45,6 +60,128 @@ public IntuneAppProtectionPolicyRequiredException(final String errorCode,
+                    originalParameters.getHomeAccountId()
+            );
+        }
+
+        setAccountUserId(uId);
+
+        final Authority authority = originalParameters.getAuthority();
+        setAuthorityUrl(authority.getAuthorityURL().toString());
+
+        final String homeAccountId = originalParameters.getHomeAccountId();
+        String tenantId = null;
+
+        if (homeAccountId != null) {
+            final Pair<String, String> tenantInfo = StringUtil.getTenantInfo(homeAccountId);
+            tenantId = tenantInfo.second;
+        }
+
+        if (TextUtils.isEmpty(tenantId) && authority instanceof AzureActiveDirectoryAuthority) {
+            tenantId = ((AzureActiveDirectoryAuthority) authority).mAudience.getTenantId();
+
+        }
+        setTenantId(tenantId);
+    }
+
+    public IntuneAppProtectionPolicyRequiredException(final String errorCode,
+                                                      final String errorMessage,
+                                                      final BrokerSilentTokenCommandParameters originalParameters) {
+        super(errorCode, errorMessage, null);
+
+        final String upn = (originalParameters.getAccountManagerAccount() != null) ?
+                originalParameters.getAccountManagerAccount().name :
+                originalParameters.getLoginHint();
+
+        setAccountUpn(upn);
+
+        String uId = originalParameters.getLocalAccountId();
+
+        if (TextUtils.isEmpty(uId)) {
+            com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Local account id is empty, attempting get user id from home account id");
+            uId = getUIdFromHomeAccountId(
+                    originalParameters.getHomeAccountId()
+            );
+        }
+
+        setAccountUserId(uId);
+
+        final Authority authority = originalParameters.getAuthority();
+        setAuthorityUrl(authority.getAuthorityURL().toString());
+
+        final String homeAccountId = originalParameters.getHomeAccountId();
+        String tenantId = null;
+
+        if (homeAccountId != null) {
+            final Pair<String, String> tenantInfo = StringUtil.getTenantInfo(homeAccountId);
+            tenantId = tenantInfo.second;
+        }
+
+        if (TextUtils.isEmpty(tenantId) && authority instanceof AzureActiveDirectoryAuthority) {
+            tenantId = ((AzureActiveDirectoryAuthority) authority).mAudience.getTenantId();
+
+        }
+        setTenantId(tenantId);
+    }
+
+    /**
+     * Helper method to get uid from home account id
+     * V2 home account format : <uid>.<utid>
+     * V1 : it's stored as <uid>
+     *
+     * @param homeAccountId
+     * @return valid uid or null if it's not in either of the format.
+     */
+    @Nullable
+    private String getUIdFromHomeAccountId(@Nullable String homeAccountId) {
+        //TODO: This method is from BrokerOperationParameterUtils...
+        // seems like this is not broker specific per se and should move to somewhere better
```

Comment:
seems like this todo is now resolved? Do we want it in this class though? Seems like it needs to become `public` and in go into some util class